sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -410,6 +410,12 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
       case _ => None
     }
 
-    filters.flatMap(translate)
+    filters.map(eliminateCastOnAttribute(_)).flatMap(translate)
+  }
+
+  private def eliminateCastOnAttribute(expression: Expression): Expression = {
+    expression.transform {
+      case Cast(a: Attribute, _) => a
+    }
```

Comment:
Thanks, I didn't notice the assumption of parquetFilters until you pointed out. I'll follow your advice above.